internal/handlers/publishers.go

```diff
@@ -58,14 +58,20 @@ func (p *Publisher) GetPublisher(ctx *fiber.Ctx) error {
 
 // PostPublisher creates a new publisher.
 func (p *Publisher) PostPublisher(ctx *fiber.Ctx) error {
-	publisher := new(requests.Publisher)
+	publisherReq := requests.Publisher{}
 
-	if err := ctx.BodyParser(&publisher); err != nil {
-		return common.Error(fiber.StatusBadRequest, "can't create Publisher", "invalid json")
+	if err := ctx.BodyParser(&publisherReq); err != nil {
+		return common.Error(fiber.StatusBadRequest, "can't create Publisher", "invalid json", err)
 	}
 
-	if err := common.ValidateStruct(*publisher); err != nil {
-		return common.Error(fiber.StatusUnprocessableEntity, "can't create Publisher", "invalid format")
+	if err := common.ValidateStruct(&publisherReq); err != nil {
+		return common.Error(fiber.StatusBadRequest, "can't create Publisher", "invalid json", err)
```

Comment:
This is a validation error, so we should use a 422 (`fiber.StatusUnprocessableEntity`) as we were originally.

Reply:
Sorry, this was a mistake. Will correct it